miden-lib/src/faucets/mod.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{assembler::assembler, auth::AuthScheme};
+use miden_objects::{
+    accounts::{Account, AccountCode, AccountId, AccountStorage, AccountType, AccountVault},
+    assembly::ModuleAst,
+    assets::TokenSymbol,
+    crypto::merkle::MerkleStore,
+    utils::{string::ToString, vec},
+    AccountError, Felt, StarkField, Word, ZERO,
+};
+
+const MAX_MAX_SUPPLY: u64 = 1 << 63;
```

Comment:
nit: the naming is slightly inconsistent here: if it is `MAX_MAX_SUPPLY` then the value should be `(1 << 63) - 1)`. I would change it here to this and then use strict inequality on line 50 below.